pkg/hive/install.go

```diff
@@ -0,0 +1,195 @@
+	if err != nil {
+		return err
+	}
+
+	psSecret, err := pullsecretSecret(doc.OpenShiftCluster.Properties.HiveProfile.Namespace, doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	cd := c.clusterDeploymentForInstall(doc, version, c.env.IsLocalDevelopmentMode())
+
+	// Enrich the cluster deployment with the correlation data so that logs are
+	// properly annotated
+	err = utillog.EnrichHiveWithCorrelationData(cd, doc.CorrelationData)
+	if err != nil {
+		return err
+	}
+	err = utillog.EnrichHiveWithResourceID(cd, doc.OpenShiftCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	resources := []kruntime.Object{
+		sppSecret,
+		envSecret(doc.OpenShiftCluster.Properties.HiveProfile.Namespace, c.env.IsLocalDevelopmentMode()),
+		psSecret,
+		installConfigCM(doc.OpenShiftCluster.Properties.HiveProfile.Namespace, doc.OpenShiftCluster.Location),
+		cd,
+	}
+
+	err = dynamichelper.Prepare(resources)
+	if err != nil {
+		return err
+	}
+
+	err = c.dh.Ensure(ctx, resources...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func servicePrincipalSecretForInstall(oc *api.OpenShiftCluster, sub *api.SubscriptionDocument, isDevelopment bool) (*corev1.Secret, error) {
+	clusterSPBytes, err := clusterSPToBytes(sub, oc)
+	if err != nil {
+		return nil, err
+	}
+
+	enc, err := json.Marshal(oc)
+	if err != nil {
+		return nil, err
+	}
+
+	encSub, err := json.Marshal(sub)
+	if err != nil {
+		return nil, err
+	}
+
+	sppSecret := clusterServicePrincipalSecret(oc.Properties.HiveProfile.Namespace, clusterSPBytes)
+	sppSecret.Data["99_aro.json"] = enc
+	sppSecret.Data["99_sub.json"] = encSub
+
+	if isDevelopment {
+		// In development mode, load in the proxy certificates so that clusters
+		// can be accessed from a local (not in Azure) Hive
+		proxyCert, err := ioutil.ReadFile("secrets/proxy.crt")
+		if err != nil {
+			return nil, err
+		}
+
+		proxyClientCert, err := ioutil.ReadFile("secrets/proxy-client.crt")
+		if err != nil {
+			return nil, err
+		}
+
+		proxyClientKey, err := ioutil.ReadFile("secrets/proxy-client.key")
+		if err != nil {
+			return nil, err
+		}
+
+		sppSecret.Data["proxy.crt"] = proxyCert
+		sppSecret.Data["proxy-client.crt"] = proxyClientCert
+		sppSecret.Data["proxy-client.key"] = proxyClientKey
+	}
+
+	return sppSecret, nil
+}
+
+func (c *clusterManager) clusterDeploymentForInstall(doc *api.OpenShiftClusterDocument, version *api.OpenShiftVersion, isDevelopment bool) *hivev1.ClusterDeployment {
+	var envVars = []corev1.EnvVar{
+		{
+			Name:  "ARO_UUID",
+			Value: doc.ID,
+		},
+	}
+
+	if isDevelopment {
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  "ARO_RP_MODE",
+			Value: "development",
+		})
+		for _, i := range devEnvVars {
+			envVars = append(envVars, makeEnvSecret(i))
+		}
+	} else {
+		for _, i := range prodEnvVars {
+			envVars = append(envVars, makeEnvSecret(i))
+		}
+	}
+
+	// Do not set InfraID here, as Hive wants to do that
```

Comment:
We need to generate it, since we use it for IDs of various network things that the cluster needs to install. In the future, we would specify it via manifests.